src/nn/context.ts

```diff
@@ -5,60 +5,99 @@ import * as wasm from '@tensorflow/tfjs-backend-wasm';
 
 import * as tf from '@tensorflow/tfjs-core';
 
+import { MLGraph, MLNamedArrayInputs, MLNamedArrayOutputs } from './graph';
 import * as utils from './utils';
 
+
+/** @internal */
+export enum MLContextType {
+  'default' = 'default',
+  'webgpu' = 'webgpu'
+}
+
+
 /**
- * [spec](https://webmachinelearning.github.io/webnn/#enumdef-mlpowerpreference)
+ * [spec](https://webmachinelearning.github.io/webnn/#enumdef-mldevicetype)
  */
-export enum MLPowerPreference {
-  'default' = 'default',
-  'high-performance' = 'high-performance',
-  'low-power' = 'low-power'
+export enum MLDeviceType {
+  'cpu' = 'cpu',
+  'gpu' = 'gpu'
 }
 
+
 /**
- * [spec](https://webmachinelearning.github.io/webnn/#enumdef-mldevicepreference)
+ * [spec](https://webmachinelearning.github.io/webnn/#enumdef-mlpowerpreference)
  */
-export enum MLDevicePreference {
+export enum MLPowerPreference {
   'default' = 'default',
-  'gpu' = 'gpu',
-  'cpu' = 'cpu'
+  'high-performance' = 'high-performance',
+  'low-power' = 'low-power'
 }
 
 
 /**
  * [spec](https://webmachinelearning.github.io/webnn/#dictdef-mlcontextoptions)
  */
 export interface MLContextOptions {
-  /** */
+  deviceType?: MLDeviceType;
   powerPreference?: MLPowerPreference;
-  /** */
-  devicePreference?: MLDevicePreference;
 }
 
 /**
  * [API spec](https://webmachinelearning.github.io/webnn/#mlcontext)
  */
 export class MLContext {
   private options_: MLContextOptions;
+  private type_: MLContextType;
 
   /** @internal */
   // eslint-disable-next-line @typescript-eslint/no-empty-function
   constructor(options: MLContextOptions = {}) {
     utils.assert(options instanceof Object, 'Invalid options.');
+    if (options.deviceType !== undefined) {
+      utils.assert(
+          options.deviceType in MLDeviceType,
+          'Invalid device type.');
+    }      
     if (options.powerPreference !== undefined) {
       utils.assert(
           options.powerPreference in MLPowerPreference,
           'Invalid power preference.');
     }
     this.options_ = options;
+    this.type_ = MLContextType.default;
   }
 
   /** @internal */
   get options(): MLContextOptions {
     return this.options_;
   }
 
+  /** @internal */
+  get type(): MLContextType {
+    return this.type_;
+  }
+
+  /**
+   * [spec](https://webmachinelearning.github.io/webnn/#dom-mlcontext-compute)
+   */
+  async compute(
+      graph: MLGraph,
+      inputs: MLNamedArrayInputs,
+      outputs: MLNamedArrayOutputs): Promise<void> {
+    await graph.compute(inputs, outputs);
+  }
+
+  /**
+   * [spec](https://webmachinelearning.github.io/webnn/#dom-mlcontext-computesync)
+   */
+  computeSync(
+      graph: MLGraph,
+      inputs: MLNamedArrayInputs,
+      outputs: MLNamedArrayOutputs): void {
+      graph.computeSync(inputs, outputs);
```

Comment:
computeSync should only be allowed in dedicated worker, please add a check for that restriction.